pkg/controller/autoscaling/elasticsearch/autoscaler/recommender/cpu.go

```diff
@@ -0,0 +1,133 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License 2.0;
+// you may not use this file except in compliance with the Elastic License 2.0.
+
+package recommender
+
+import (
+	"fmt"
+
+	"github.com/go-logr/logr"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+
+	esv1 "github.com/elastic/cloud-on-k8s/v2/pkg/apis/elasticsearch/v1"
+	"github.com/elastic/cloud-on-k8s/v2/pkg/controller/autoscaling/elasticsearch/resources"
+	"github.com/elastic/cloud-on-k8s/v2/pkg/controller/autoscaling/elasticsearch/status"
+	"github.com/elastic/cloud-on-k8s/v2/pkg/controller/elasticsearch/client"
+)
+
+type cpu struct {
+	base
+	requiredNodeCPUCapacity, requiredTotalCPUCapacity *client.AutoscalingCapacity
+}
+
+func (m *cpu) HasResourceRecommendation() bool {
+	return true
+}
+
+func (m *cpu) ManagedResource() corev1.ResourceName {
+	return corev1.ResourceCPU
+}
+
+func (m *cpu) NodeResourceQuantity() resource.Quantity {
+	max := m.autoscalingSpec.CPURange.Max.MilliValue()
+	// Surface the situation where CPU is exhausted.
+	if m.requiredNodeCPUCapacity.MilliValue() > max {
+		// Elasticsearch requested more CPU per node than allowed by the user
+		err := fmt.Errorf("node required CPU is greater than the maximum one")
```

Comment:
I'm not sure but it sounds weird.
```suggestion
		err := fmt.Errorf("CPU required by the node is greater than the maximum allowed")
```

Reply:
required or requested?

Reply:
> required or requested?

required: https://www.elastic.co/guide/en/elasticsearch/reference/current/autoscaling-get-autoscaling-capacity.html#autoscaling-get-autoscaling-capacity-api-response-body